src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreUnconfiguredInputDataSource.cs

```diff
@@ -40,13 +47,22 @@ protected override IDisposable LinkExternalInput(ITargetBlock<RestoreInfo> targe
                 JoinableFactory,
                 _project);
 
-            // Transform all restore data -> combined restore data
-            DisposableValue<ISourceBlock<RestoreInfo>> mergeBlock = joinBlock.TransformWithNoDelta(update => update.Derive(MergeRestoreInputs));
+            IPropagatorBlock<IProjectVersionedValue<RestoreValues>, RestoreInfo> transformBlock
+                = DataflowBlockSlim.CreateTransformBlock<IProjectVersionedValue<RestoreValues>, RestoreInfo>(update => update.Derive(MergeRestoreInputs));
+
+            // Publish when all target frameworks get loaded
+            var mergeBlock = ProjectDataSources.SyncLinkTo(
+                _loadedActiveConfiguredProjectsDataSource.SourceBlock.SyncLinkOptions(),
+                joinBlock.SyncLinkOptions(),
+                target: transformBlock, 
+                linkOptions: DataflowOption.PropagateCompletion,
+                cancellationToken: CancellationToken.None
+                );
 
             JoinUpstreamDataSources(_activeConfigurationGroupService.ActiveConfiguredProjectGroupSource);
```

Comment:
Do you also have to join `_loadedActiveConfiguredProjectsDataSource` here?

```suggestion
            JoinUpstreamDataSources(_activeConfigurationGroupService.ActiveConfiguredProjectGroupSource, _loadedActiveConfiguredProjectsDataSource);
```